Added a slide on visitor pattern with lambdas (#239)
hsf-training/cpluspluscourse: talk/morelanguage/lambda.tex

    \begin{itemize}
    \item they are replaced by a functor before compilation
    \end{itemize}
    \begin{columns}
      \scriptsize
      \begin{column}{.25\textwidth}
        \begin{cppcode*}{gobble=6}
          int sum = 0, off = 1;
          auto l =
          [&sum, off]

          (int x) {
            sum += x + off;
          };

          l(42);
        \end{cppcode*}
      \end{column}
      \begin{column}{.45\textwidth}
        \begin{cppcode*}{gobble=6, firstnumber=21}
          int sum = 0, off = 1;
          struct __lambda4 {
            int& sum;
            int off;
            __lambda4(int& s, int o)
            : sum(s), off(o) {}
            auto operator()(int x)const{
              sum += x + off;
            }
          };
          auto l = __lambda4{sum, off};
          l(42);
        \end{cppcode*}
      \end{column}
    \end{columns}
  \end{block}
  \begin{exampleblock}{Some nice consequence}
    \begin{itemize}
    \item lambdas are regular objects

>     \item they can in particular inherit from each other !

Reviewer: ```suggestion
    \item they can in particular be inherited from!
```

  \end{alertblock}
\end{frame}

\begin{frame}[fragile]

>   \frametitlecpp[17]{lambdas and the visitor pattern}

Reviewer: I wonder whether the slide should be moved to the ones about `std::variant`.

Author: Well, at that stage we did not yet cover lambdas, so it would be weird.

Author: issue #246 was open concerning a possible refactoring of the slides there. And we've decided to merge this as is for the moment
